liblouis/maketable.c

```diff
@@ -95,9 +95,25 @@ printRule(TranslationTableRule *rule, widechar *rule_string) {
 		rule_string[l++] = ' ';
 		for (int k = 0; k < rule->charslen; k++) rule_string[l++] = rule->charsdots[k];
 		rule_string[l++] = ' ';
-		for (int k = 0; k < rule->dotslen; k++)
-			rule_string[l++] = _lou_getCharFromDots(
+		for (int k = 0; k < rule->dotslen; k++) {
+			rule_string[l] = _lou_getCharFromDots(
 					rule->charsdots[rule->charslen + k], displayTable);
+			if (rule_string[l] == '\0') {
+				// if a dot pattern can not be displayed, print an error message
+				char *message = (char *)malloc(50 * sizeof(char));
+				sprintf(message, "ERROR: provide a display rule for %s",
```

Comment:
Aha, I thought you were going to crap out here with `_lou_logMessage` just like in the other cases. But I guess this is ok too.

Reply:
Well it turned out it it's not so easy to do in the case of lou_maketable. I think this is a good enough solution.